### DISCUSSION TITLE
What is the clustering used in Pooling?

### DISCUSSION THREAD

**Discussion by hkim716** [pyg-team/pytorch_geometric#2271] [Q&A]
I'm trying to understand some pooling layers `max_pool` or `mean_pool` in PyG, but I'm stuck in the concept of `clustering`.
Why do we need the clustering when we coarsen the graphs?
Is it because we do not know node connections for the coarsened graphs?

**Comment by rusty1s** [comment]
The pooling strategy of using clusters of nodes was first introduced [here](https://arxiv.org/abs/1606.09375), and it dictates how you want to coarsen the graph, i.e., nodes in the same cluster will form a super node in the coarsened graph. Supernodes are connected if there exists nodes in the underlying cluster which were connected in the original graph.

**Reply by hkim716** [reply]
Thanks for the information Matt. 
I have another question that I got from someone. For 2d-convolution with image data, user can specify `the numbers of channels` and `the size of the channels`. However, in `GCNConv`, I only see the number of channels, which defines `the number of node features` at the next layer.  Why I cannot see any `size of channels` in GCNConv layer? Is that because `Message passing` mechanism is totally different from image datasets?

**Reply by rusty1s** [reply]
You mean kernel size? The `GCNConv` transforms each neighbor equally, so there does not exist such an argument here. However, there exists some GNNs that try to simulate such behaviour, e.g., see our `torch_geometric.nn.conv.SplineConv` operator.
